module/testmodule/fakemodule.sh

```diff
@@ -1,7 +1,10 @@
 #!/bin/bash
 
-# fakemodule is a completely fake module that echos a message and exits. Used
-# to test that modules that never respond to ready requests will not be
-# restarted.
+# fakemodule is a completely fake module that repeatedly echos a mesasge. Used
+# to test that modules that never respond to ready requests will be stopped.
 
-echo "this is a fake module; exiting now"
+while :
+do
+  echo "fakemodule is running"
+  sleep 0.01
+done
```

Comment:
should we still have a case where the module immediately crashes?

Reply:
Good q. IIUC, to the module manager, there is no difference between an immediate crash and the above code. They both represent an inability of the module to take ownership over the `.sock` file, and therefore result in a `timed out waiting for module to start listening` error. I modified inline comments to clarify that `fakemodule.sh` never starts listening. We do not currently have a test for a module that takes ownership over the `.sock` file but does not respond to a ready request. I can add that, but it should follow almost the same code path.

Reply:
does the mod manager do anything different based on if the process is still alive?

Reply:
Not that I know of. `stopProcess` will return `nil` early in the case that the process is nil (already stopped).